spec/loophp/collection/Iterator/TypedIteratorSpec.php

```diff
@@ -29,7 +29,7 @@ public function it_allows_array_of_any_type(): void
     {
         $data = [self::MAP_DATA, self::LIST_DATA];
 
-        $this->beConstructedWith($data);
+        $this->beConstructedWith(new ArrayIterator($data));
```

Comment:
why do we not want to allow this to be constructed from `array`?

Reply:
1. We avoid mixing types.
2. I rather use the decorator pattern, it seems to be the way to go here.